platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/DefaultProjectComponentSelector.java

```diff
@@ -100,7 +105,15 @@ public ImmutableAttributes getAttributes() {
 
     @Override
     public List<Capability> getRequestedCapabilities() {
-        return requestedCapabilities;
+        return capabilitySelectors.stream()
+            .filter(c -> c instanceof SpecificCapabilitySelector)
+            .map(c -> ((DefaultSpecificCapabilitySelector) c).getBackingCapability())
```

Comment:
💭 Seeing this conversion again really suggests to me that you should be able to ask a selector to convert itself to a capability.

Reply:
Resolving this comment, as we can discuss as part of a related comment